sdk/core/Azure.Core/api/Azure.Core.net461.cs

```diff
@@ -762,14 +763,16 @@ public TelemetryDetails(System.Reflection.Assembly assembly, string? application
         public void Apply(Azure.Core.HttpMessage message) { }
         public override string ToString() { throw null; }
     }
-    public abstract partial class TokenCredential
+    public abstract partial class TokenCredential : System.ClientModel.CredentialProvider
     {
         protected TokenCredential() { }
+        public override System.ClientModel.Credential GetCredential(System.Collections.Generic.IReadOnlyDictionary<string, object> context, System.Threading.CancellationToken cancellationToken) { throw null; }
```

Comment:
This terminology makes the API hard to understand. 'Token credential" is a credential and it's strange that it has a method called "GetCredential"